File: factorial.py
while(1):
    num=int(input("enter the number"))
    if num==0:
        print("0 factorial is 1\n")
    elif num>0:
        fact =1
        for x in range(num,0,-1):
            fact = fact*x
        print(num,"factorial is",fact)
    else:
        print("invalid number\n")
    ch=input("Do you want to continue:(y/n)")
    #print(ch)
    if ch=="y" or ch=="Y":
        continue
    else:
        break

File: factorial.cpp
#include<iostream>
using namespace std;
int main()
{
	int i,num,fact;
	char ch;
	do
	{
		cout<<"Enter any positive Integer:";
		cin>>num;

		if(num==0)
		{
			cout<<"0 factorial is 1"<<endl;
		}	
		else if(num>0)
		{
			fact=1;
			for(i=num;i>=1;i--)
			{
				fact=fact*i;
			}
			cout<<num<<" factorial is "<<fact<<endl;
		}
		else
		{
			cout<<"Invalid number"<<endl;
		}
		cout<<"Do you want to check another number:(y/n)";
		cin>>ch;
	}while(ch=='y'||ch=='Y');
	//printf("%d factorial is %d\n",num,fact);
	return 0;
}
